storybook/stories/test_cases/21_domain_edges.story.tsx

```diff
@@ -0,0 +1,219 @@
+ * Side Public License, v 1.
+ */
+
+import { boolean, number } from '@storybook/addon-knobs';
+import React from 'react';
+
+import {
+  LineSeries,
+  Axis,
+  BarSeries,
+  Chart,
+  ScaleType,
+  Settings,
+  Position,
+  HistogramBarSeries,
+  niceTimeFormatter,
+} from '@elastic/charts';
+import { getRandomNumberGenerator } from '@elastic/charts/src/mocks/utils';
+
+import { ChartsStory } from '../../types';
+import { useBaseTheme } from '../../use_base_theme';
+
+const rng = getRandomNumberGenerator('chart');
+const start = Date.parse('2023-07-01T00:00:00.000Z').valueOf();
+const interval = 1000 * 60 * 5;
+const data = Array.from({ length: 20 }, (d, i) => {
+  return [start + interval * i, Math.floor(rng(2, 10))];
+});
+
+export const Example: ChartsStory = (_, { title, description }) => {
+  const customDomain = boolean('custom domain', false);
+  const startTimeSlider = number('start time', 0, { min: 0, max: 20, range: true, step: 0.2 });
+  const rangeSlider = number('end time', 19, { min: 1, max: 25, range: true, step: 0.2 });
+  const subtract = boolean('subtract 1ms', false);
+
+  const tickFormat = (d: number) => new Date(d).toISOString();
+  const xDomain = customDomain
+    ? { min: start + interval * startTimeSlider, max: start + interval * rangeSlider - (subtract ? 1 : 0) }
+    : undefined;
+  const domain: [number, number] = [xDomain?.min ?? data[0][0] ?? 0, xDomain?.max ?? data.at(-1)?.[0] ?? 0];
+  const limitedData = data
+    .filter((d) => d[0] <= (xDomain?.max ?? Infinity))
+    .map((d, i) => [d[0], i === 10 ? null : d[1]]);
+
+  return (
+    <>
+      <p>
+        {xDomain ? 'configured' : 'data'} domain: {new Date(domain[0]).toISOString()} to{' '}
+        {new Date(domain[1]).toISOString()}
+      </p>
+      <Chart title={title} description={description} size={['100%', 250]}>
+        <Settings
+          baseTheme={useBaseTheme()}
+          showLegend
+          legendPosition={{
+            floatingColumns: 1,
+            floating: true,
+            direction: 'vertical',
+            hAlign: 'right',
+            vAlign: 'top',
+          }}
+          showLegendExtra
+          xDomain={xDomain}
+          theme={{
+            scales: {
+              histogramPadding: 0,
+              barsPadding: 0,
+            },
+            barSeriesStyle: {
+              rectBorder: {
+                visible: true,
+                strokeWidth: 0,
+                stroke: 'white',
+              },
+            },
+          }}
+        />
+        <Axis
+          id="x"
+          tickFormat={tickFormat}
+          position={Position.Bottom}
+          gridLine={{ visible: true }}
+          timeAxisLayerCount={2}
+          style={{
+            tickLabel: {
+              alignment: { horizontal: 'left' },
+              padding: 0,
+            },
+            tickLine: {
+              padding: 0,
+              size: 5,
+            },
```

Comment:
With the new eui theme adoption, the `tickLine.visible` property is `false`. So with multilayer time axes I add it explicitly.

```suggestion
            tickLine: {
              visible: true,
              padding: 0,
              size: 5,
            },
```

Reply:
fixed in https://github.com/elastic/elastic-charts/pull/2115/commits/6af36e5d71d0adc0fbbb643837888f58df5b35b2